src/9d10_len_1080.c

```diff
@@ -1,7 +1,106 @@
 #include "common.h"
 #include "camera.h"
 
-INCLUDE_ASM(void, "9d10_len_1080", update_camera_mode_4, Camera* camera);
+void update_camera_mode_4(Camera* camera) {
+    f32 sinBoom;
+    f32 cosBoom;
+    f32 deltaX;
+    f32 deltaY;
+    f32 deltaZ;
+    f32 deltaX2;
+    f32 deltaY2;
+    f32 deltaZ2;
+    f32 boomYaw;
+    f32 boomLength;
+    f32 f0;
+    f32 f4;
+    f32 new_var;
+
+    f0 = camera->targetPos.x;
+    deltaX = camera->auxPitch;
+    if (f0 > deltaX) {
+        f0 = deltaX;
+    }
+    if (f0 < -deltaX) {
+        f0 = -deltaX;
+    }
+    camera->lookAt_obj_target.x = f0;
+    f4 = camera->targetPos.z;
+    deltaZ = camera->auxBoomLength;
+    if (f4 > deltaZ) {
+        f4 = deltaZ;
+    }
+    if (f4 < -deltaZ) {
+        f4 = -deltaZ;
+    }
+    camera->lookAt_obj_target.z = f4;
+    camera->unk_70 = 0.0f;
+    camera->currentBoomYaw = 0.0f;
+    camera->trueRotation.x = camera->unk_70;
+    camera->currentBoomLength = camera->lookAt_dist * D_8009A5EC;
+    camera->currentYOffset = camera->auxBoomPitch * D_8009A5EC;
+    if (camera->unk_06 != 0) {
+        camera->unk_06 = 0;
+        camera->unk_98 = 0.0f;
+        camera->unk_9C = 0.0f;
+        camera->lookAt_obj.x = camera->lookAt_obj_target.x;
+        camera->lookAt_obj.y = camera->lookAt_obj_target.y + camera->currentYOffset;
+        camera->lookAt_obj.z = camera->lookAt_obj_target.z;
+        boomYaw = DEG_TO_RAD(camera->currentBoomYaw);
+        sinBoom = sin_rad(boomYaw);
+        cosBoom = cos_rad(boomYaw);
+        deltaY = deltaX = 0.0f;
+        deltaZ = camera->currentBoomLength;
+        deltaX2 = deltaX;
```

Comment:
Does it still match if you set these to 0.0 instead of chaining them? It would be cleaner if so.

Reply:
yee, 
x = 0
y = 0 would look nicer imo